packages/client/src/Dashboard.test.tsx

```diff
@@ -0,0 +1,120 @@
+import Conversation from "./Conversation"
+import Dashboard from "./Dashboard"
+import { mount, updates } from "./testing"
+import * as $ from "./testing/fixtures"
+
+it("stars a conversation in list view", async () => {
+  const app = mount(<Dashboard accountId={$.account.id} />, {
+    mocks: [$.getAccountMock, $.flagMock]
+  })
+
+  await updates(app)
+  app.find(Avatar).simulate("click")
+  app.find('button[aria-label="star"]').simulate("click")
+  await updates(app, 10)
+
+  expect(app.find(ListItemText)).toIncludeText("⭐ ")
+})
+
+it("un-stars a conversation in list view", async () => {
+  const app = mount(<Dashboard accountId={$.account.id} />, {
+    mocks: [
+      {
+        ...$.getAccountMock,
+        result: {
+          data: {
+            account: {
+              ...$.account,
+              conversations: [{ ...$.conversation, isStarred: true }]
+            }
+          }
+        }
+      },
+      {
+        ...$.flagMock,
+        request: {
+          ...$.flagMock.request,
+          variables: { ...$.flagMock.request.variables, isFlagged: true }
+        },
+        result: {
+          data: {
+            conversations: {
+              ...$.flagMock.result.data.conversations,
+              flag: { ...$.conversation, isStarred: false }
+            }
+          }
+        }
+      }
+    ]
+  })
+
+  await updates(app)
+  app.find(Avatar).simulate("click")
+  app.find('button[aria-label="star"]').simulate("click")
+  await updates(app, 10)
+
+  expect(app.find(ListItemText)).not.toIncludeText("⭐ ")
+})
+
+it("stars a conversation in conversation view", async () => {
+  const app = mount(
+    <Conversation
+      accountId={$.account.id}
+      conversationId={$.conversation.id}
+    />,
+    {
+      mocks: [$.getConversationMock, $.flagMock]
+    }
+  )
+
+  await updates(app)
+  app.find('button[aria-label="star"]').simulate("click")
+  await updates(app, 10)
+
+  expect(app.find(StarBorder)).toExist()
+})
+
+it("un-stars a conversation in conversation view", async () => {
+  const app = mount(
+    <Conversation
+      accountId={$.account.id}
+      conversationId={$.conversation.id}
+    />,
+    {
+      mocks: [
+        {
+          ...$.getConversationMock,
+          result: {
+            data: { conversation: { ...$.conversation, isStarred: true } }
+          }
+        },
+        {
+          ...$.flagMock,
+          request: {
+            ...$.flagMock.request,
+            variables: { ...$.flagMock.request.variables, isFlagged: true }
+          },
+          result: {
+            data: {
+              conversations: {
+                ...$.flagMock.result.data.conversations,
+                flag: { ...$.conversation, isStarred: false }
+              }
+            }
+          }
+        }
+      ]
+    }
+  )
+
+  await updates(app)
+  app.find('button[aria-label="star"]').simulate("click")
+  await updates(app, 10)
+
+  expect(app.find(StarIcon)).toExist()
+})
```

Comment:
These are good tests! I would like to add some tests for starring and unstarring multiple conversations at once. I noticed that when multiple conversations are selected if only some are starred then clicking the star action removes the stars. The behavior is nicely consistent - but I would like it to be flipped so that clicking the star action adds stars unless all of the selected conversations are starred.